src/libraries/System.Private.Uri/src/System/Uri.cs

```diff
@@ -1648,36 +1651,7 @@ public override bool Equals(object? comparand)
             if (NotAny(Flags.AllUriInfoSet) || obj.NotAny(Flags.AllUriInfoSet))
             {
                 // Try raw compare for _strings as the last chance to keep the working set small
-                if (!IsUncOrDosPath)
-                {
-                    if (_string.Length == obj._string.Length)
-                    {
-                        unsafe
-                        {
-                            // Try case sensitive compare on _strings
-                            fixed (char* selfPtr = _string)
-                            {
-                                fixed (char* otherPtr = obj._string)
-                                {
-                                    // This will never go negative since _string is checked to be a valid URI
-                                    int i = (_string.Length - 1);
-                                    for (; i >= 0; --i)
-                                    {
-                                        if (*(selfPtr + i) != *(otherPtr + i))
-                                        {
-                                            break;
-                                        }
-                                    }
-                                    if (i == -1)
-                                    {
-                                        return true;
-                                    }
-                                }
-                            }
-                        }
-                    }
-                }
-                else if (string.Equals(_string, obj._string, StringComparison.OrdinalIgnoreCase))
+                if (string.Equals(_string, obj._string, IsUncOrDosPath ? StringComparison.OrdinalIgnoreCase : StringComparison.Ordinal))
```

Comment:
Is it faster for the common non-UncOrDosPath case to do:
```C#
if (!IsUncOrDosPath
{
    if (string.Equals(_string, obj._string))
    {
        return true;
    }
}
else if (string.Equals(_string, obj._string, StringComparison.OrdinalIgnoreCase)
{
    return true;
}
```
or does it not make a meaningful difference?

Reply:
I don't see any improvement.